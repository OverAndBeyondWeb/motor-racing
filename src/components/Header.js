import React from 'react';
import { Link } from 'react-router-dom';

const Header = ({ title }) => (
  <header ClassName="Header">
    <div className="title">
      <h1>Names</h1>
    </div>
    <nav className="main-nav">
      <ul>
        <Link to="/names"><li>Race Names</li></Link>
        <Link to="/locations"><li>Race Locations</li></Link>
      </ul>
    </nav>
  </header>
);

export default Header;