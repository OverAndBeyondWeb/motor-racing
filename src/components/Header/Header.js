import React from 'react';
import { Link } from 'react-router-dom';

const Header = ({ title }) => (
  <header className="Header">
    <div className="title">
      <h1>{title}</h1>
    </div>
    <nav className="main-nav">
      <ul>
        <Link to="/names"><li>Race Names</li></Link>
        <Link to="/locations"><li>Race Locations</li></Link>
      </ul>
    </nav>
  </header>
);

export default Header;