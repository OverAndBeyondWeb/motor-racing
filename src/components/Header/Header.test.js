import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';

import { shallow } from 'enzyme';

describe('<App />', () => {

  const header = shallow(<Header />);

  test('renders without crashing', () => {
    expect(header).toMatchSnapshot();
  });

});
