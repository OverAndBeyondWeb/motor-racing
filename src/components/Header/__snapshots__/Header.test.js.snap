// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          <h1 />
        </div>,
        <nav
          className="main-nav"
        >
          <ul>
            <Link
              replace={false}
              to="/names"
            >
              <li>
                Race Names
              </li>
            </Link>
            <Link
              replace={false}
              to="/locations"
            >
              <li>
                Race Locations
              </li>
            </Link>
          </ul>
        </nav>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1 />,
          "className": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul>
            <Link
              replace={false}
              to="/names"
            >
              <li>
                Race Names
              </li>
            </Link>
            <Link
              replace={false}
              to="/locations"
            >
              <li>
                Race Locations
              </li>
            </Link>
          </ul>,
          "className": "main-nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/names"
              >
                <li>
                  Race Names
                </li>
              </Link>,
              <Link
                replace={false}
                to="/locations"
              >
                <li>
                  Race Locations
                </li>
              </Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <li>
                  Race Names
                </li>,
                "replace": false,
                "to": "/names",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Race Names",
                },
                "ref": null,
                "rendered": "Race Names",
                "type": "li",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <li>
                  Race Locations
                </li>,
                "replace": false,
                "to": "/locations",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Race Locations",
                },
                "ref": null,
                "rendered": "Race Locations",
                "type": "li",
              },
              "type": [Function],
            },
          ],
          "type": "ul",
        },
        "type": "nav",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            <h1 />
          </div>,
          <nav
            className="main-nav"
          >
            <ul>
              <Link
                replace={false}
                to="/names"
              >
                <li>
                  Race Names
                </li>
              </Link>
              <Link
                replace={false}
                to="/locations"
              >
                <li>
                  Race Locations
                </li>
              </Link>
            </ul>
          </nav>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1 />,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul>
              <Link
                replace={false}
                to="/names"
              >
                <li>
                  Race Names
                </li>
              </Link>
              <Link
                replace={false}
                to="/locations"
              >
                <li>
                  Race Locations
                </li>
              </Link>
            </ul>,
            "className": "main-nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="/names"
                >
                  <li>
                    Race Names
                  </li>
                </Link>,
                <Link
                  replace={false}
                  to="/locations"
                >
                  <li>
                    Race Locations
                  </li>
                </Link>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <li>
                    Race Names
                  </li>,
                  "replace": false,
                  "to": "/names",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Race Names",
                  },
                  "ref": null,
                  "rendered": "Race Names",
                  "type": "li",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <li>
                    Race Locations
                  </li>,
                  "replace": false,
                  "to": "/locations",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Race Locations",
                  },
                  "ref": null,
                  "rendered": "Race Locations",
                  "type": "li",
                },
                "type": [Function],
              },
            ],
            "type": "ul",
          },
          "type": "nav",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
